src/components/DataListSection.tsx

```diff
@@ -77,36 +77,49 @@ export const DataListSection = ({
     skipLastDivider: true
   });
 
+  if (listData?.length) {
+    return (
+      <View>
+        {!!sectionTitle && (
+          <SectionHeader onPress={navigate} title={sectionTitle ?? getTitleForQuery(query)} />
+        )}
+        {!!limit &&
+          (listData?.length ? (
+            <ListComponent
+              data={isRandom ? _shuffle(listData).slice(0, limit) : listData.slice(0, limit)}
+              horizontal={horizontal}
+              navigation={navigation}
+              query={query}
+            />
+          ) : (
+            !!placeholder && placeholder
+          ))}
+        {!!linkTitle && !!navigateLink && showLink && (
+          <Wrapper>
+            <Touchable onPress={navigateLink}>
+              <BoldText center primary underline>
+                {linkTitle}
+              </BoldText>
+            </Touchable>
+          </Wrapper>
+        )}
+        {!!buttonTitle && !!navigateButton && showButton && (
+          <Wrapper>
+            <Button title={buttonTitle} onPress={navigateButton} />
+          </Wrapper>
+        )}
+      </View>
+    );
+  }
+
+  if (!placeholder) {
+    return null;
+  }
+
   return (
     <View>
-      {!!sectionTitle && (
-        <SectionHeader onPress={navigate} title={sectionTitle ?? getTitleForQuery(query)} />
-      )}
-      {!!limit &&
-        (listData?.length ? (
-          <ListComponent
-            data={isRandom ? _shuffle(listData).slice(0, limit) : listData.slice(0, limit)}
-            horizontal={horizontal}
-            navigation={navigation}
-            query={query}
-          />
-        ) : (
-          !!placeholder && placeholder
-        ))}
-      {!!linkTitle && !!navigateLink && showLink && (
-        <Wrapper>
-          <Touchable onPress={navigateLink}>
-            <BoldText center primary underline>
-              {linkTitle}
-            </BoldText>
-          </Touchable>
-        </Wrapper>
-      )}
-      {!!buttonTitle && !!navigateButton && showButton && (
-        <Wrapper>
-          <Button title={buttonTitle} onPress={navigateButton} />
-        </Wrapper>
-      )}
+      <SectionHeader onPress={navigate} title={sectionTitle ?? getTitleForQuery(query)} />
```

Comment:
In line 83 there is a `!!sectionTitle` before rendering the section header. Do we need it also here?
I think we can probably remove it in line 83 as well, right, as we have `sectionTitle ?? getTitleForQuery(query)`, so there is some fallback implemented for the prop?

Reply:
Yes, this check is unnecessary.

Reply:
can be removed at line 61 as well, right?

Reply:
https://github.com/ikuseiGmbH/smart-village-app-app/pull/583/commits/c572a17d9f6c37319dec493afd2b8a6a9c43d631

Reply:
what about 61?

```
{!!sectionTitle && (
    <SectionHeader onPress={navigate} title={sectionTitle ?? getTitleForQuery(query)} />
  )}
```

Reply:
I realised after I shared the code. I'll fix it.